Core/include/Acts/Navigation/MultiLayerNavigationPolicy.hpp

```diff
@@ -0,0 +1,168 @@
+        << m_volume.volumeName());
+    const Transform3& itransform = m_volume.transform().inverse();
+    const Acts::Vector3 locPosition = itransform * args.position;
+    const Acts::Vector3 locDirection = itransform * args.direction;
+
+    std::vector<Vector2> path = generatePath(locPosition, locDirection);
+
+    const auto& surfaces = m_volume.surfaces();
+    std::vector<const Surface*> surfCandidates = {};
+    surfCandidates.reserve(surfaces.size());
+
+    for (const auto& pos : path) {
+      std::vector<std::size_t> indices = m_indexedGrid.grid.atPosition(pos);
+
+      std::ranges::transform(indices, std::back_inserter(surfCandidates),
+                             [&](const auto& i) { return &surfaces[i]; });
+    }
+
+    // remove duplicated candidates
+
+    ACTS_VERBOSE("MultiLayerNavigationPolicy Candidates reported"
+                 << surfCandidates.size() << " candidates");
+
+    // fill the navigation stream with the container
+    for (const auto* surf : surfCandidates) {
+      stream.addSurfaceCandidate(*surf, args.tolerance);
+    }
+  }
+
+  /// Connect this policy with a navigation delegate
+  /// @param delegate The navigation delegate to connect to
+  void connect(NavigationDelegate& delegate) const override {
+    connectDefault<MultiLayerNavigationPolicy>(delegate);
+  }
+
+  /// Generate a path in the multilayer
+  /// @param startPosition The starting position of the path (in local frame)
+  /// @param direction The direction of the path (in local frame)
+  /// @return A vector of positions along the path
+  std::vector<Vector2> generatePath(const Vector3& startPosition,
+                                    const Vector3& direction) const {
+    std::vector<Vector2> path;
+
+    auto maxXIndex = m_indexedGrid.grid.numLocalBins()[0];
+    auto maxYIndex = m_indexedGrid.grid.numLocalBins()[1];
+    Vector3 unitDir = direction.normalized();
+
+    for (std::size_t i = 0; i < maxYIndex; i++) {
+      auto v1 = m_indexedGrid.grid.lowerLeftBinEdge({1, i + 1});
+      auto v2 = m_indexedGrid.grid.upperRightBinEdge({maxXIndex, i + 1});
+
+      auto intersection = detail::IntersectionHelper2D::intersectSegment(
+          Vector2(v1[0], v1[1]), Vector2(v2[0], v2[1]),
+          startPosition.template block<2, 1>(0, 0),
+          unitDir.template block<2, 1>(0, 0));
+      if (!intersection.isValid()) {
+        continue;
+      }
+
+      path.push_back(intersection.position());
+    }
+    return path;
+  }
+
+ private:
+  // The tracking volume
+  const TrackingVolume& m_volume;
+
+  // The grid that holds the indexed surfaces
+  indexed_grid m_indexedGrid;
+};
+
+// Make alias for the static assert check for the Navigation Policy
+using gridType =
+    Grid<std::vector<std::size_t>,
+         Axis<AxisType::Equidistant, Acts::AxisBoundaryType::Bound>,
+         Axis<AxisType::Equidistant, Acts::AxisBoundaryType::Bound>>;
+using indexedUpdatorType = Experimental::IndexedSurfacesNavigation<gridType>;
```

Comment:
```suggestion
using IndexedUpdatorType = Experimental::IndexedSurfacesNavigation<gridType>;
```
forwarding `Experimental::` to `Acts::` looks a bit scary